x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_metrics.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Client } from '@elastic/elasticsearch';
+import type { IndexedHostsResponse } from './index_endpoint_hosts';
+import { EndpointDocGenerator } from '../generate_data';
+
+import { wrapErrorAndRejectPromise } from './utils';
+
+/**
+ * Indexes the requested number of documents for the endpoint host metrics currently being output by the generator.
+ *
+ *
+ *
+ * @param numDocs
+ * @param client
+ * @param metricsIndex
+ * @param generator
+ * @param indexedHosts
+ */
+export async function indexEndpointMetricsDocs({
+  numDocs,
+  client,
+  metricsIndex,
+  generator,
+  indexedHosts,
+}: {
+  numDocs: number;
+  client: Client;
+  metricsIndex: string;
+  generator: EndpointDocGenerator;
+  indexedHosts: IndexedHostsResponse;
+}): Promise<void> {
```

Comment:
Can this indexer return back the list of documents created?

Also,

Can you add a `deleteIndexedEndpointMetricsDocs()` method that takes in the return value of this function and cleans up deletes the data.

You can look at the other `index*` modules here to see the pattern.